src/map/status.cpp

```diff
@@ -9265,18 +9263,14 @@ int status_change_start(struct block_list* src, struct block_list* bl,enum sc_ty
 
 	// Check failing SCs from list
 	if (!scdb->fail.empty()) {
-		// Don't let OPT1 that have RemoveOnDamaged start a new effect
-		if (sc->opt1 > OPT1_NONE && scdb->flag[SCF_REMOVEONDAMAGED]) {
-			unit_data *ud = unit_bl2ud(bl);
-
-			if (ud != nullptr && ud->lastEffect == type) {
-				ud->lastEffect = SC_NONE;
+		for (const auto &it : scdb->fail) {
+			// Don't let OPT1 that have RemoveOnDamaged start a new effect in the same attack.
+			if (sc->lastEffect == it) {
+				sc->lastEffect = SC_NONE;
```

Comment:
Setting sc->lastEffect to SC_NONE should also happen if the status does not fail.

In fact it would be better if this is reset after all the status changes were applied (just in case there is a skill that applies two status changes, there's also certain cards that can apply multiple status changes).

If we move it elsewhere you can simplify the code here to:
```if (sc->data[it] || sc->lastEffect == it)
  return 0;```